.github/workflows/ci-model-regression.yml

```diff
@@ -649,102 +648,68 @@ jobs:
           export PR_URL="https://github.com/${GITHUB_REPOSITORY}/pull/${{ github.event.number }}"
           poetry run pip install analytics-python
           poetry run python .github/scripts/mr_publish_results.py
-          poetry run python .github/scripts/mr_generate_summary.py
           cat $SUMMARY_FILE
 
       - name: Upload an artifact with the report
         uses: actions/upload-artifact@v2
         if: steps.set_dataset_config_vars.outputs.is_dataset_exists == 'true' && steps.set_dataset_config_vars.outputs.is_config_exists == 'true'
         with:
-          name: report.json
-          path: ./report.json
+          name: report-${{ matrix.dataset }}-${{ matrix.config }}
+          path: report.json
 
       - name: Stop Datadog Agent
         if: steps.set_dataset_config_vars.outputs.is_dataset_exists == 'true' && steps.set_dataset_config_vars.outputs.is_config_exists == 'true'
         run: |
           sudo service datadog-agent stop
 
-  add_comment_results_gpu:
-    name: Add a comment with the results
+  combine_reports:
+    name: Combine reports
     runs-on: ubuntu-latest
     needs:
+      - model_regression_test_cpu
       - model_regression_test_gpu
-
+    if: always() && (needs.model_regression_test_cpu.result == 'success' || needs.model_regression_test_gpu.result == 'success')
```

Comment:
I think `always()` is not needed. if a result for the `model_regression_test_cpu` is not successful then it will be skipped anyway.

Reply:
I think it is needed, because:

I would like to run `combine_reports` if either `model_regression_test_cpu` or `model_regression_test_gpu` is successful.

I added `model_regression_test_cpu` to the `needs` section. We expect either `model_regression_test_cpu` or `model_regression_test_gpu` to run. 
By default, I think `needs` works in a way that both would have to be successful (https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#jobsjob_idneeds), but in our case only one will be successful and the other will be skipped. 

Similar discussion: https://github.com/actions/runner/issues/491#issuecomment-850884422